### PR TITLE
deep copy of ComplexCommand

### PR DESCRIPTION
**Pull request by bartekmaka06**

### CODE REVIEW

**Review by kowallus — CHANGES_REQUESTED** [review]
The solution looks fine. There is one issue which requires your attention.

**Comment by bartekmaka06** [comment]
Is it copy method in DriverCommandManager class?

File: src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java [hunk 1]
```diff
@@ -48,6 +48,11 @@ public void accept(ICommandVisitor visitor) {
 				visitor.visit(this);
 			}
 
+			@Override
+			public DriverCommand copy() {
+				return null;
+			}
+
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java]: You shouldn't simply add code not working correctly. Maybe e.g., it is possible to replace anonymous ICompoundCommand() with ComplexCommand implementation.

**Comment by kowallus** [comment]
Yes. I added a missing comment (I think I forgot to confirm it)

File: src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java [hunk 2]
```diff
@@ -9,10 +9,17 @@ public class ComplexCommand implements ICompoundCommand {
 
     private final List<DriverCommand> commandList;
 
+    String name=null;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/ComplexCommand.java]: should be private!
I suggest avoiding this null value. Is public ComplexCommand(List<DriverCommand> commandList) constructor necessary?

**Review by kowallus — CHANGES_REQUESTED** [review]
Please be more careful

**Review by kowallus — APPROVED** [review]

**Pull request merged**